msal4j-sdk/src/main/java/com/microsoft/aad/msal4j/ServiceFabricManagedIdentitySource.java

```diff
@@ -0,0 +1,89 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.microsoft.aad.msal4j;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.Collections;
+import java.util.HashMap;
+
+class ServiceFabricManagedIdentitySource extends AbstractManagedIdentitySource {
+
+    private static final Logger LOG = LoggerFactory.getLogger(ServiceFabricManagedIdentitySource.class);
+
+    private static final String SERVICE_FABRIC_MSI_API_VERSION = "2019-07-01-preview";
+
+    private final URI msiEndpoint;
+    private final String identityHeader;
+    private final ManagedIdentityIdType idType;
+    private final String userAssignedId;
+
+    @Override
+    public void createManagedIdentityRequest(String resource) {
+        managedIdentityRequest.baseEndpoint = msiEndpoint;
+        managedIdentityRequest.method = HttpMethod.GET;
+
+        managedIdentityRequest.headers = new HashMap<>();
+        managedIdentityRequest.headers.put("secret", identityHeader);
+
+        managedIdentityRequest.queryParameters = new HashMap<>();
+        managedIdentityRequest.queryParameters.put("resource", Collections.singletonList(resource));
+        managedIdentityRequest.queryParameters.put("api-version", Collections.singletonList(SERVICE_FABRIC_MSI_API_VERSION));
+
+        if (idType == ManagedIdentityIdType.CLIENT_ID) {
+            LOG.info("[Managed Identity] Adding user assigned client id to the request.");
```

Comment:
nit: mention the source name in these log messages.

Reply:
Added in the latest commit